### PR TITLE
Remove circular references from error objects

### PR DESCRIPTION
**Pull request by ivov**
This PR removes circular references from error objects to prevent `JSON.stringify()` from throwing a conversion error when post-execution hooks are fired.

Relates to #1767

### CODE REVIEW

File: packages/workflow/src/NodeErrors.ts [hunk 1]
```diff
@@ -132,9 +141,33 @@ abstract class NodeError extends Error {
...
+				value.forEach((val, index) => {
+					if (seen.has(val)) {
+						value[index] = '[circular]';
+						return;
+					}
+					if (this.isTraversableObject(val)) {
+						this.removeCircularRefs(val, seen);
+					}
+				});
+			}
```
Comment (BHesseldieck) [inline packages/workflow/src/NodeErrors.ts:168]: we don't handle nested arrays here, do we keep it out of scope to not further add complexity? @ivov

Comment (BHesseldieck) [reply packages/workflow/src/NodeErrors.ts]: it's not outdated 🙈

Comment (BHesseldieck) [reply packages/workflow/src/NodeErrors.ts]: decided to keep it out of scope for now

**Comment by janober** [comment]
Thanks! Got merged.

**Comment by janober** [comment]
Got released with n8n@0.124.0

**Pull request merged**